src/main/java/org/commcare/formplayer/session/MenuSession.java

```diff
@@ -210,20 +211,43 @@ public boolean autoAdvanceMenu(Screen screen, boolean autoAdvanceMenu)
     }
 
     private void addTitle(String input, Screen previousScreen) {
-        if (previousScreen instanceof EntityScreen) {
-            try {
-                String caseName = SessionUtils.tryLoadCaseName(sandbox.getCaseStorage(), input);
+        if (previousScreen instanceof MultiSelectEntityScreen) {
+            UUID referenceId = ((MultiSelectEntityScreen)previousScreen).getStorageReferenceId();
+            String[] values = entitiesSelectionCache.read(referenceId);
```

Comment:
This is no longer correct. We now store the full instance in the DB:
```java
VirtualDataInstance instance = virtualDataInstanceService.read(referenceId);
TreeElement root = instance.getRoot();
```

You'll probably have to evaluate an xpath expression (or you could just use TreeElement methods like 'getNumChildren' and 'getChildAt').

Reply:
ah, I was working off outdated code. Updated in 15034ca9c00318843df954c6df7a4870374ad2e0, and exposed a `getVirtualInstance` method in https://github.com/dimagi/commcare-core/pull/1105